.github/workflows/spelling.yml

```diff
@@ -7,24 +41,82 @@ on:
 jobs:
   spelling:
     name: Spell checking
+    permissions:
+      contents: read
+      pull-requests: read
+      actions: read
+      security-events: write
+    outputs:
+      followup: ${{ steps.spelling.outputs.followup }}
     runs-on: ubuntu-latest
+    if: "contains(github.event_name, 'pull_request') || github.event_name == 'push'"
+    concurrency:
+      group: spelling-${{ github.event.pull_request.number || github.ref }}
+      # note: If you use only_check_changed_files, you do not want cancel-in-progress
+      cancel-in-progress: true
     steps:
-    - name: checkout-merge
-      if: "contains(github.event_name, 'pull_request')"
-      uses: actions/checkout@v2
-      with:
-        ref: refs/pull/${{github.event.pull_request.number}}/merge
-    - name: checkout
-      if: ${{ github.event_name == 'push' ||
-        (
-          contains(github.event.comment.body, '@check-spelling-bot apply')
-        ) }}
-      uses: actions/checkout@v2
-    - uses: check-spelling/check-spelling@main
+    - name: check-spelling
       id: spelling
-      if: ${{ github.event_name != 'issue_comment' ||
-        (
-          contains(github.event.comment.body, '@check-spelling-bot apply')
-        ) }}
+      uses: check-spelling/check-spelling@main
+      with:
+        suppress_push_for_open_pull_request: 1
+        checkout: true
+        post_comment: 0
+        experimental_apply_changes_via_bot: 1
+        dictionary_source_prefixes: |
+          {"cspell": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20220816/dictionaries/"}
```

Comment:
The `@prerelease` version of check spelling is using a newer cspell dictionary set and their file paths changed between `@main` and `@prerelease`. As `check_extra_dictionaries` is emptied, it's simpler to just specify the newer version here than either find the older paths or add a mapping for the old and new.